.github/workflows/main.yml

```diff
@@ -0,0 +1,129 @@
+name: Build and Upload story Binary
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  # TESTING ONLY
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  id-token: write
+  contents: write
+  pull-requests: write
+  actions: write
+
+env:
+  NUM_INTERNAL_BINARIES_TO_KEEP: 50
+  NUM_PUBLIC_BINARIES_TO_KEEP: 400
+  S3_BUCKET: story-geth-binaries
+  BIN_NAME: story
+
+jobs:
+  Timestamp:
+    uses: storyprotocol/gha-workflows/.github/workflows/reusable-timestamp.yml@main
+  
+  build_and_push:
+    needs: Timestamp
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: [linux-amd64]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+
+      - name: Authenticate to GCP
+        uses: google-github-actions/auth@v1
+        with:
+          workload_identity_provider: ${{ secrets.GCP_WORKLOAD_IDENTITY_PROVIDER_REPO_STORY }}
+          service_account: ${{ secrets.GCP_SERVICE_ACCOUNT_REPO_STORY }}
+
+      - name: Set up Google Cloud SDK
+        uses: google-github-actions/setup-gcloud@v1
+
+      - name: Extract the version
+        run: |
+          PARAMS_FILE="./lib/buildinfo/buildinfo.go"
+          VERSION_MAJOR=$(awk -F= '/VersionMajor/ {gsub(/[^0-9]/, "", $2); printf "%s", $2}' $PARAMS_FILE)
+          VERSION_MINOR=$(awk -F= '/VersionMinor/ {gsub(/[^0-9]/, "", $2); printf "%s", $2}' $PARAMS_FILE)
+          VERSION_PATCH=$(awk -F= '/VersionPatch/ {gsub(/[^0-9]/, "", $2); printf "%s", $2}' $PARAMS_FILE)
+          VERSION_META=$(awk -F\" '/VersionMeta/ {print $2; exit}' $PARAMS_FILE)
+
+          VERSION="$VERSION_MAJOR.$VERSION_MINOR.$VERSION_PATCH"
+          if [ "$VERSION_META" != "stable" ]; then
+            VERSION+="-${VERSION_META}"
+          fi
+
+          echo "Version extracted: $VERSION"
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+          echo "VERSION_META=$VERSION_META" >> $GITHUB_ENV
+
+      - name: Build the story binary
+        run: |
+          IFS="-" read -r GOOS GOARCH <<< "${{ matrix.platform }}"
+
+          echo "Building for $GOOS/$GOARCH..."
+          cd client
+          env GOOS=$GOOS GOARCH=$GOARCH go build -o $BIN_NAME -v || {
```

Comment:
I think writing `env GOOS=$GOOS GOARCH=$GOARCH go build ...` is effectively the same as:
```bash
GOOS=$GOOS GOARCH=$GOARCH go build -o $BIN_NAME -v
```